### PR TITLE
[nest] Migrate to bnd build

### PR DESCRIPTION
**Pull request by wborn**
The migration is done but the tests fail because for some reason the Service provided by the `@Component` annotated `NestTestHandlerFactory` cannot be found. This results in:

```
TEST handleSetPointCelsiusCommands(org.openhab.binding.nest.handler.NestThermostatHandlerTest) <<< ERROR: Could not get NestTestHandlerFactory
Expected: is not null
     but: was null
java.lang.AssertionError: Could not get NestTestHandlerFactory
Expected: is not null
     but: was null
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.openhab.binding.nest.handler.NestThingHandlerOSGiTest.setUp(NestThingHandlerOSGiTest.java:155)
```

Do you know how to get this working @maggu2810 ? I can imagine it is a bit special because this one of the few tests that uses its own `@Component` annotated classes.

### CODE REVIEW

**Comment by J-N-K** [comment]
The DMX tests use somthing similar. I can't see any obvious difference.

**Comment by wborn** [comment]
I think the main difference is that `DmxTestHandlerFactory` is no longer used and a left over from the times those tests were extending from `JavaOSGiTests instead of `JavaTest`. It build just fine when I remove the class. :wink:

**Comment by J-N-K** [comment]
Good point. I‘ll remove it.

**Comment by wborn** [comment]
The tests now work again by manually registering the `NestTestHandlerFactory` instead of using an `@Component` annotation.

File: addons/binding/org.openhab.binding.globalcache/OSGI-INF/org.openhab.binding.globalcache.internal.GlobalCacheHandlerFactory.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.globalcache/OSGI-INF/org.openhab.binding.globalcache.internal.GlobalCacheHandlerFactory.xml]: These look misplaced here?

File: addons/extensionservice/org.openhab.extensionservice.marketplace.automation/OSGI-INF/org.openhab.extensionservice.marketplace.automation.internal.AutomationExtensionHandler.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (martinvw) [inline addons/extensionservice/org.openhab.extensionservice.marketplace.automation/OSGI-INF/org.openhab.extensionservice.marketplace.automation.internal.AutomationExtensionHandler.xml]: So do these?

**Comment by wborn** [comment]
Seems my Eclipse is again messing up PRs. :-)

**Comment by J-N-K** [comment]
The problem arises when bindings are moved to the new build after they have been built locally. On checkout all files are moved, except those that are not in git. Most are suppressed by global `.gitignore` but the ones in `OSGI-INF` have their own ignore-file which is removed during the movement.

**Comment by J-N-K** [comment]
LGTM, I would prefer to see network merged before, I already rebased that two times and this will create a new merge conflict :-)

**Comment by martinvw** [comment]
@J-N-K that is so mean of you now @wborn has to fix his merge conflicts instead.

**Comment by wborn** [comment]
I've fixed the merge conflicts for the 4th time! :disappointed:

**Comment by martinvw** [comment]
@wborn did you really fix it yet another time :-D

**Comment by wborn** [comment]
Yes I inflicted the conflict on myself by merging other PRs. :smile:

**Pull request merged**